File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int numIterations = max(word1.length(), word2.length());
        string result = "";
        for (int i=0; i<numIterations;i++){
            if (i<word1.length()){
                result += word1[i];
            }
            if (i<word2.length()){
                result += word2[i];
            }
        }
        return result;
    }
};

File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        result = ""
        for i in range(min(len(word1), len(word2))):
            result += word1[i] + word2[i]
        if len(word1) < len(word2):
            result += word2[len(word1):]
        else:
            result += word1[len(word2):]
        return result

